packages/@stylexjs/babel-plugin/__tests__/when-functions-test.js

```diff
@@ -0,0 +1,280 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+jest.autoMockOff();
+
+import { transformSync } from '@babel/core';
+import stylexPlugin from '../src/index';
+
+function transform(source, opts = {}) {
+  const result = transformSync(source, {
+    filename: opts.filename ?? 'test.js',
+    parserOpts: {
+      flow: 'all',
+    },
+    babelrc: false,
+    plugins: [[stylexPlugin, { ...opts }]],
+  });
+
+  return result;
+}
+
+describe('@stylexjs/babel-plugin', () => {
+  describe('[transform] when functions', () => {
+    test('when.ancestor function', () => {
+      const { code, metadata } = transform(`
+        import { when, create } from '@stylexjs/stylex';
+        
+        const styles = create({
+            container: {
+              backgroundColor: {
+                default: 'blue',
+                [when.ancestor(':hover')]: 'red',
+              },
+            },
+        });
+
+        console.log(styles.container);
+      `);
+
+      expect(code).toMatchInlineSnapshot(`
+        "import { when, create } from '@stylexjs/stylex';
+        const styles = {
+          container: {
+            kWkggS: "x1t391ir x26ppzc",
+            $$css: true
+          }
+        };
+        console.log(styles.container);"
+      `);
+
+      expect(metadata.stylex).toMatchInlineSnapshot(`
+        [
+          [
+            "x1t391ir",
+            {
+              "ltr": ".x1t391ir{background-color:blue}",
+              "rtl": null,
+            },
+            3000,
+          ],
+          [
+            "x26ppzc",
+            {
+              "ltr": ".x26ppzc:where(.stylex-target:hover *){background-color:red}",
```

Comment:
@necolas @fabiospampinato 

Example of the actual selector being generated so far.